File: src/main/java/org/pixeltime/enchantmentsenhance/events/enchantments/Farmer.java
package org.pixeltime.enchantmentsenhance.events.enchantments;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.BlockPlaceEvent;
import org.bukkit.inventory.ItemStack;
import org.pixeltime.enchantmentsenhance.manager.SettingsManager;

public class Farmer implements Listener {
    @EventHandler
    public void onPlace(final BlockPlaceEvent blockPlaceEvent) {
        final String translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantments." + "farmer"));
        final Player player = blockPlaceEvent.getPlayer();
        try {
            ItemStack[] armorContents;
            for (int length = (armorContents = player.getInventory().getArmorContents()).length, i = 0; i < length; ++i) {
                final ItemStack itemStack = armorContents[i];
                if (itemStack.hasItemMeta() && itemStack.getItemMeta().hasLore() && itemStack.getItemMeta().getLore().contains(String.valueOf(translateAlternateColorCodes) + " I")) {
                    if (player.getItemInHand().getType() == Material.CARROT_ITEM) {
                        blockPlaceEvent.getBlockPlaced().setTypeIdAndData(Material.CARROT.getId(), (byte) 7, true);
                    }
                    if (player.getItemInHand().getType() == Material.MELON_SEEDS) {
                        blockPlaceEvent.getBlockPlaced().setType(Material.MELON);
                    }
                    if (player.getItemInHand().getType() == Material.POTATO_ITEM) {
                        blockPlaceEvent.getBlockPlaced().setTypeIdAndData(Material.POTATO.getId(), (byte) 7, true);
                    }
                    if (player.getItemInHand().getType() == Material.SEEDS) {
                        blockPlaceEvent.getBlockPlaced().setTypeIdAndData(Material.CROPS.getId(), (byte) 7, true);
                    }
                    if (player.getItemInHand().getType() == Material.PUMPKIN_SEEDS) {
                        blockPlaceEvent.getBlockPlaced().setType(Material.PUMPKIN);
                    }
                }
            }
        } catch (Exception ex) {
        }
    }
}

File: src/main/kotlin/org/pixeltime/enchantmentsenhance/event/enchantments/Farmer.kt
package org.pixeltime.enchantmentsenhance.event.enchantment

import org.bukkit.ChatColor
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.BlockPlaceEvent
import org.bukkit.inventory.ItemStack
import org.pixeltime.enchantmentsenhance.manager.SettingsManager

class Farmer : Listener {
    @EventHandler
    fun onPlace(blockPlaceEvent: BlockPlaceEvent) {
        val translateAlternateColorCodes = ChatColor.translateAlternateColorCodes('&', SettingsManager.lang.getString("enchantment." + "farmer"))
        val player = blockPlaceEvent.player
        try {
            val armorContents = player.inventory.armorContents
            val length = armorContents.size
            var i = 0
            while (i < length) {
                val itemStack = armorContents[i]
                if (itemStack.hasItemMeta() && itemStack.itemMeta.hasLore() && itemStack.itemMeta.lore.contains(translateAlternateColorCodes.toString() + " I")) {
                    if (player.itemInHand.type == Material.CARROT_ITEM) {
                        blockPlaceEvent.blockPlaced.setTypeIdAndData(Material.CARROT.id, 7.toByte(), true)
                    }
                    if (player.itemInHand.type == Material.MELON_SEEDS) {
                        blockPlaceEvent.blockPlaced.type = Material.MELON
                    }
                    if (player.itemInHand.type == Material.POTATO_ITEM) {
                        blockPlaceEvent.blockPlaced.setTypeIdAndData(Material.POTATO.id, 7.toByte(), true)
                    }
                    if (player.itemInHand.type == Material.SEEDS) {
                        blockPlaceEvent.blockPlaced.setTypeIdAndData(Material.CROPS.id, 7.toByte(), true)
                    }
                    if (player.itemInHand.type == Material.PUMPKIN_SEEDS) {
                        blockPlaceEvent.blockPlaced.type = Material.PUMPKIN
                    }
                }
                ++i
            }
        } catch (ex: Exception) {
        }

    }
}
